src/itype.h

```diff
@@ -420,7 +420,10 @@ struct islot_magazine {
     int reload_time = 100;
 
     /** For ammo belts one linkage (of given type) is dropped for each unit of ammo consumed */
-     itype_id linkage = "NULL";
+    itype_id linkage = "NULL";
+
+    /** If false, ammo will cook off if this mag is affected by fire */
+    bool protects_contents = false;
```

Comment:
Better named `insulated`. Eventually this should be a scalar as opposed to boolean value specifying the degree of protection.